tutorials/hyperspectral/RasterStack-RGB-Images-in-R-Using-HSI.Rmd

```diff
@@ -109,67 +108,53 @@ the file structure.
-# we want to save our files
-#setwd("working-dir-path-here")
+# we want to save our files. Be sure to move the download into your working directory!
+wd="~/Desktop/Hyperspectral_Tutorial/" #This will depend on your local environment
+setwd(wd)
 
-```
+# create path to file name
+f <- paste0(wd,"NEON_hyperspectral_tutorial_example_subset.h5")
 
-```{r read-in-file }
+```
 
-# Read in H5 file
-f <- 'NEON-DS-Imaging-Spectrometer-Data.h5'
+```{r view-file-structure, eval=FALSE, comment=NA}
 #View HDF5 file structure 
-h5ls(f,all=T)
+View(h5ls(f,all=T))
 
 ```
 
 To spatially locate our raster data, we need a few key attributes:
 
 * The coordinate reference system
-* The lower left hand corner X, Y location of the raster
-* The dimensions (number of pixels in the x and y directions) of the raster to 
-   use to calculate the **extent**.
+* The spatial extent of the raster
    
 We'll begin by grabbing these key attributes from the H5 file.   
 
 ```{r get-spatial-attributes }
 
-# R get spatial info and map info using the h5readAttributes function
-spInfo <- h5readAttributes(f,"spatialInfo")
-# define coordinate reference system
-myCrs <- spInfo$projdef
-# define the resolution
-res <- spInfo$xscale
-
-#Populate the raster image extent value. 
-mapInfo<-h5read(f,"map info")
-# the map info string contains the lower left hand coordinates of our raster
-# let's grab those next
-# split out the individual components of the mapinfo string
-mapInfo<-unlist(strsplit(mapInfo, ","))
-
-# grab the utm coordinates of the lower left corner
-xMin<-as.numeric(mapInfo[4])
-yMax<-as.numeric(mapInfo[5]) 
-
-# R get attributes for the Reflectance dataset
-reflInfo <- h5readAttributes(f,"Reflectance")
-
-#create objects represents the dimensions of the Reflectance dataset
-#note that there are several ways to access the size of the raster contained
-#within the H5 file
-nRows <- reflInfo$row_col_band[1]
-nCols <- reflInfo$row_col_band[2]
-nBands <- reflInfo$row_col_band[3]
-
-#grab the no data value
-myNoDataValue <- reflInfo$`data ignore value`
-myNoDataValue
+# define coordinate reference system from the EPSG code provided in the HDF5 file
+myEPSG <- h5read(f,"/SJER/Reflectance/Metadata/Coordinate_System/EPSG Code" )
+myCRS <- crs(paste0("+init=epsg:",myEPSG))
+
+# get the Reflectance_Data attributes
+reflInfo <- h5readAttributes(f,"/SJER/Reflectance/Reflectance_Data" )
+
+# Grab the UTM coordinates of the spatial extent
+xMin <- reflInfo$Spatial_Extent_meters[1]
+xMax <- reflInfo$Spatial_Extent_meters[2]
+yMin <- reflInfo$Spatial_Extent_meters[3]
+yMax <- reflInfo$Spatial_Extent_meters[4]
+
+# define the extent (left, right, top, bottom)
+rasExt <- extent(xMin,xMax,yMin,yMax)
+# view the extent to make sure that it looks right
+rasExt
 
+# Finally, define the no data value for later
+myNoDataValue <- as.integer(reflInfo$Data_Ignore_Value)
+myNoDataValue
 ```
 
 Next, we'll write a function that will perform the processing that we did step by 
```

Comment:
Line 162: Can you replace what is linked with the actual title: Intro to Working with Hyperspectral Remote Sensing Data in HDF5 Format in R

Reply:
Yes, done.